skaffold init can't be interrupted with ctrl-c

yeah I actually get something even worse when I run this on the `microservices` example:

```
➜  microservices git:(master) ✗ skaffold init
? Choose the builder to build image gcr.io/k8s-skaffold/leeroy-app  [Use arrows to move, space to select, type to filter]
> Docker (leeroy-app/Dockerfile)
  Docker (leeroy-web/Dockerfile)
  None (image not built from these sources)
? Choose the builder to build image gcr.io/k8s-skaffold/leeroy-web  [Use arrows to move, space to select, type to filter]
> Docker (leeroy-app/Dockerfile)
  Docker (leeroy-web/Dockerfile)
  None (image not built from these sources)
WARN[0002] unused builder configs found in repository: [{leeroy-app/Dockerfile} {leeroy-web/Dockerfile}]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x51f7cfd]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer.processBuildArtifacts(0xc000824100, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	pkg/skaffold/initializer/init.go:383 +0xfd
github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer.generateSkaffoldConfig(0x5897e60, 0xc0002041e0, 0xc000824100, 0x2, 0x2, 0x2, 0x2, 0x2, 0x2, 0xc0002d2940)
	pkg/skaffold/initializer/init.go:402 +0x137
github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer.DoInit(0x58c33e0, 0xc000343940, 0x588c800, 0xc0000e8000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	pkg/skaffold/initializer/init.go:167 +0x3da
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doInit(0x58c33e0, 0xc000343940, 0x588c800, 0xc0000e8000, 0xc0003023e0, 0xc000030f00)
	cmd/skaffold/app/cmd/init.go:55 +0x151
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.cancelWithCtrlC.func1(0x588c800, 0xc0000e8000, 0x0, 0x0)
	cmd/skaffold/app/cmd/signals.go:33 +0xba
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.(*builder).NoArgs.func1(0xc0002f8280, 0x68560a8, 0x0, 0x0, 0x0, 0x0)
	cmd/skaffold/app/cmd/commands.go:96 +0x6b
github.com/spf13/cobra.(*Command).execute(0xc0002f8280, 0x68560a8, 0x0, 0x0, 0xc0002f8280, 0x68560a8)
	vendor/github.com/spf13/cobra/command.go:826 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc000030f00, 0xc0000e8000, 0x588c800, 0xc0000e8008)
	vendor/github.com/spf13/cobra/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	vendor/github.com/spf13/cobra/command.go:864
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x588c800, 0xc0000e8000, 0x588c800, 0xc0000e8008, 0xc0004b3f88, 0x400781f)
	cmd/skaffold/app/skaffold.go:27 +0x58
main.main()
	cmd/skaffold/skaffold.go:30 +0x4e
```

--- a/pkg/skaffold/initializer/init.go
+++ b/pkg/skaffold/initializer/init.go
@@ -162,7 +162,11 @@ func DoInit(ctx context.Context, out io.Writer, c Config) error {
 			}
 			pairs = append(pairs, newPairs...)
 		} else {
-			pairs = append(pairs, resolveBuilderImages(unresolvedBuilderConfigs, unresolvedImages)...)
+			resolved, err := resolveBuilderImages(unresolvedBuilderConfigs, unresolvedImages)
+			if err != nil {
+				return err
+			}
+			pairs = append(pairs, resolved...)
 		}
 	}
 
@@ -321,10 +325,10 @@ func processCliArtifacts(artifacts []string) ([]builderImagePair, error) {
 }
 
 // For each image parsed from all k8s manifests, prompt the user for the builder that builds the referenced image
-func resolveBuilderImages(builderConfigs []InitBuilder, images []string) []builderImagePair {
+func resolveBuilderImages(builderConfigs []InitBuilder, images []string) ([]builderImagePair, error) {
 	// If nothing to choose, don't bother prompting
 	if len(images) == 0 || len(builderConfigs) == 0 {
-		return []builderImagePair{}
+		return []builderImagePair{}, nil
 	}
 
 	// if we only have 1 image and 1 build config, don't bother prompting
@@ -332,7 +336,7 @@ func resolveBuilderImages(builderConfigs []InitBuilder, images []string) []build
 		return []builderImagePair{{
 			Builder:   builderConfigs[0],
 			ImageName: images[0],
-		}}
+		}}, nil
 	}
 
 	// Build map from choice string to builder config struct
@@ -351,8 +355,13 @@ func resolveBuilderImages(builderConfigs []InitBuilder, images []string) []build
 		if len(images) == 0 {
 			break
 		}
+
 		image := images[0]
-		choice := promptUserForBuildConfigFunc(image, choices)
+		choice, err := promptUserForBuildConfigFunc(image, choices)
+		if err != nil {
+			return nil, err
+		}
+
 		if choice != NoBuilder {
 			pairs = append(pairs, builderImagePair{Builder: choiceMap[choice], ImageName: image})
 			choices = util.RemoveFromSlice(choices, choice)
@@ -362,10 +371,10 @@ func resolveBuilderImages(builderConfigs []InitBuilder, images []string) []build
 	if len(choices) > 0 {
 		logrus.Warnf("unused builder configs found in repository: %v", choices)
 	}
-	return pairs
+	return pairs, nil
 }
 
-func promptUserForBuildConfig(image string, choices []string) string {
+func promptUserForBuildConfig(image string, choices []string) (string, error) {
 	var selectedBuildConfig string
 	options := append(choices, NoBuilder)
 	prompt := &survey.Select{
@@ -373,8 +382,12 @@ func promptUserForBuildConfig(image string, choices []string) string {
 		Options:  options,
 		PageSize: 15,
 	}
-	survey.AskOne(prompt, &selectedBuildConfig, nil)
-	return selectedBuildConfig
+	err := survey.AskOne(prompt, &selectedBuildConfig, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return selectedBuildConfig, nil
 }
 
 func processBuildArtifacts(pairs []builderImagePair) latest.BuildConfig {

--- a/pkg/skaffold/initializer/init_test.go
+++ b/pkg/skaffold/initializer/init_test.go
@@ -344,15 +344,16 @@ func TestResolveBuilderImages(t *testing.T) {
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
 			// Overrides promptUserForBuildConfig to choose first option rather than using the interactive menu
-			t.Override(&promptUserForBuildConfigFunc, func(image string, choices []string) string {
+			t.Override(&promptUserForBuildConfigFunc, func(image string, choices []string) (string, error) {
 				if !test.shouldMakeChoice {
 					t.FailNow()
 				}
-				return choices[0]
+				return choices[0], nil
 			})
 
-			pairs := resolveBuilderImages(test.buildConfigs, test.images)
+			pairs, err := resolveBuilderImages(test.buildConfigs, test.images)
 
+			t.CheckNoError(err)
 			t.CheckDeepEqual(test.expectedPairs, pairs)
 		})
 	}

